src/vmm/src/controller.rs

```diff
@@ -141,15 +152,16 @@ impl VmmController {
         drive_id: &str,
         disk_image: File,
     ) -> result::Result<(), DriveError> {
-        // The unwrap is safe because this is only called after the inner Vmm has booted.
-        let handler = self
-            .epoll_context
-            .get_device_handler_by_device_id::<virtio::BlockEpollHandler>(TYPE_BLOCK, drive_id)
```

Comment:
Can't wait to port the net device as well and remove this complicated function :smile: